### DISCUSSION TITLE
WiX 4 setup project cannot be opened

### DISCUSSION THREAD

**Discussion by forrma** [wixtoolset/issues#7519] [Questions] [answered]
I have converted a WiX 3 setup project to WiX 4 using the Visual Studio Extension 'HeatWave for VS2022'. The project has created a msi afterwards. Now I opened the project on another computer and the following error appears:

```
Unsupported
This version of Visual Studio is unable to open the following projects. The project types may not be installed or this version of Visual Studio may not support them.
For more information on enabling these project types or otherwise migrating your assets, please see the details in the "Migration Report" displayed after clicking OK.
	 - MyProduct.Setup, "C:\work\MyProduct\VMyProduct.Setup\MyProduct.Setup.wixproj"
```

According to the documentation no additional tool needs to be installed for WiX 4.

Thanks for hints

**Comment by forrma** [comment, marked as answer]
I found out. Visual Studio needs an extension! Try the following:

- Open Visual Studio
- Continue without code
- Extensions -> Manage Extensions -> Online -> Search 'HeatWave'.
- Install 'HeatWave for VS2022
- Close Visual Studio (the extension will be installed)
- After that you should be able to open the solution with the setup project without any errors

**Reply by BMurri** [reply]
You can mark your own response as your answer
